dask/sizeof.py

```diff
@@ -44,12 +45,16 @@ def sizeof_array(o):
 @sizeof.register(frozenset)
 def sizeof_python_collection(seq):
     num_items = len(seq)
-    samples = 10
-    if num_items > samples:
-        s = getsizeof(seq) + num_items / samples * sum(
-            map(sizeof, random.sample(seq, samples))
-        )
-        return int(s)
+    num_samples = 10
+    if num_items > num_samples:
+        if isinstance(seq, (set, frozenset)):
+            # As of Python v3.9, it is deprecated to call random.sample() on
+            # sets but since sets are unordered anyways we can simply pick
+            # the first `num_samples` items.
+            samples = itertools.islice(seq, num_samples)
```

Comment:
In theory sets are unordered, but in practice this always returns the same output for me when I naively try it with a simple set:

```python
import itertools

list(itertools.islice({3, 1, 4}, 2))
```
```

Reply:
True, it is deterministic in the sense that the same set will produce the same order, which I think is fine in this case.
My point is that sampling using `itertools.islice` is (all most) as likely to be representative as using `random.sample()`.

Reply:
I am saying _all most as likely_ because the implementation of sets could have inherent bias towards a particular order but again, I don't think it is a problem in this use case.

Reply:
Ah, neat idea @madsbk! Yeah I agree that, since sets are unordered, in practice returning the first 10 items is _probably_ just as representative as 10 random items. @jsignell do you have any objections to just using the first 10 items here?

Reply:
Yeah it seems ok to me I guess. I assumed there was a reason why we didn't just want to get the first n in the original version of this.